Kuzzle.Tests/API/Controllers/BulkControllerTest.cs

```diff
@@ -52,32 +55,34 @@ public class BulkControllerTest {
       {_id: '<otherDocumentId>', _source: {}, _version: 1, created: true}],
       total: 2}}");
 
-      JObject result = await _bulkController.mWriteAsync("foo", "bar", JArray.Parse(@"[
+      JObject result = await _bulkController.MWriteAsync("foo", "bar", JArray.Parse(@"[
           {_id: '<documentId>', body: {}},
-          {_id: '<otherDocumentId>', body:{}}]"), "wait_for", notify);
-
-      _api.Verify(new JObject {
-          {"index", "foo"},
-          {"collection", "bar"},
-          {"controller", "bulk"},
-          {"action", "mWrite"},
-          {"notify", notify},
-          {"refresh", "wait_for"},
-          {"body", JObject.Parse(@"{documents: [
-          {_id: '<documentId>', body: {}},
-          {_id: '<otherDocumentId>', body:{}}]}")}
-      });
+          {_id: '<otherDocumentId>', body:{}}]"), refresh, notify);
+
+      JObject verifyQuery = new JObject {
+          {"index", "foo"},
+          {"collection", "bar"},
+          {"controller", "bulk"},
+          {"action", "mWrite"},
+          {"body", JObject.Parse(@"{documents: [
+          {_id: '<documentId>', body: {}},
+          {_id: '<otherDocumentId>', body:{}}]}")}
+      };
+
+      QueryUtils.HandleRefreshOption(verifyQuery, refresh);
+      QueryUtils.HandleNotifyOption(verifyQuery, notify);
+
+      _api.Verify(verifyQuery);
 
       Assert.Equal<JObject>(expected, result, new JTokenEqualityComparer());
     }
   
-
     [Theory]
-    [InlineData("Some input", "foobar", null)]
-    [InlineData("", "documentId", null)]
-    [InlineData("Some input", "foobar", true)]
-    [InlineData("Some input", "foobar", false)]
-    public async void WriteAsyncTestSuccess(string documentInput, string documentId, bool? notify) {
+    [InlineData("Some input", "foobar", false, false)]
+    [InlineData("", "documentId", false, false)]
+    [InlineData("Some input", "foobar", true, false)]
+    [InlineData("Some input", "foobar", false, true)]
+    public async void WriteAsyncTestSuccess(string documentInput, string documentId, bool notify, bool refresh = false) {
```

Comment:
(nitpicking) you don't need to default the `refresh` argument

Reply:
True, the refresh argument has no default value now.